e2etest/native_staking_test.go

```diff
@@ -822,4 +825,52 @@ func TestCandidateTransferOwnership(t *testing.T) {
 			},
 		})
 	})
+	t.Run("migrate stake", func(t *testing.T) {
+		contractAddress := "io1dkqh5mu9djfas3xyrmzdv9frsmmytel4mp7a64"
+		cfg := initCfg()
+		cfg.Genesis.SystemStakingContractV2Address = contractAddress
+		cfg.Genesis.SystemStakingContractV2Height = 1
+		cfg.DardanellesUpgrade.BlockInterval = time.Second * 8640
+		test := newE2ETest(t, cfg)
+		defer test.teardown()
+
+		chainID := test.cfg.Chain.ID
+		stakerID := 1
+		stakeAmount, _ := big.NewInt(0).SetString("10000000000000000000000", 10)
+		stakeDurationDays := uint32(1) // 1day
+		candOwnerID := 2
+		blocksPerDay := 24 * time.Hour / cfg.DardanellesUpgrade.BlockInterval
+		// getBlockTime := func(height uint64) *timestamppb.Timestamp {
+		// 	t, err := test.cs.BlockTimeCalculator().CalculateBlockTime(height)
+		// 	require.NoError(err)
+		// 	return timestamppb.New(t)
+		// }
+		h := identityset.Address(1).String()
+		t.Logf("address 1: %v\n", h)
+		minAmount, _ := big.NewInt(0).SetString("100000000000000000000", 10) // 100 IOTX
+		deployCode, err := staking.StakingContractABI.Constructor.Inputs.Pack(minAmount, common.BytesToAddress(identityset.Address(stakerID).Bytes()))
+		deployCode = append(stakingContractBytecode, deployCode...)
+		require.NoError(err)
+		test.run([]*testcase{
+			{
+				name:   "deploy staking contract",
+				act:    &actionWithTime{mustNoErr(action.SignedExecution("", identityset.PrivateKey(stakerID), test.nonceMgr.pop(identityset.Address(stakerID).String()), big.NewInt(0), gasLimit, gasPrice, deployCode, action.WithChainID(chainID))), time.Now()},
+				expect: []actionExpect{successExpect, &executionExpect{contractAddress}},
+			},
+			{
+				name: "non-owner cannot migrate stake",
+				preActs: []*actionWithTime{
+					{mustNoErr(action.SignedCandidateRegister(test.nonceMgr.pop(identityset.Address(candOwnerID).String()), "cand1", identityset.Address(1).String(), identityset.Address(1).String(), identityset.Address(candOwnerID).String(), registerAmount.String(), 1, true, nil, gasLimit, gasPrice, identityset.PrivateKey(candOwnerID), action.WithChainID(chainID))), time.Now()},
+					{mustNoErr(action.SignedCreateStake(test.nonceMgr.pop(identityset.Address(stakerID).String()), "cand1", stakeAmount.String(), stakeDurationDays, true, nil, gasLimit, gasPrice, identityset.PrivateKey(stakerID), action.WithChainID(chainID))), time.Now()},
+				},
+				act:    &actionWithTime{mustNoErr(action.SignedMigrateStake(test.nonceMgr.pop(identityset.Address(2).String()), 1, gasLimit, gasPrice, identityset.PrivateKey(2), action.WithChainID(chainID))), time.Now()},
+				expect: []actionExpect{&basicActionExpect{nil, uint64(iotextypes.ReceiptStatus_ErrUnauthorizedOperator), ""}},
+			},
+			{
+				name:   "success to migrate stake",
```

Comment:
the e2e test doesn't covered the gas consumption calculation, which should be:
the balance diff of the sender account is equal to intrinsic gas + evm gas, and the value is also identical to receipt.GasConsumed

Reply:
added